staging/src/k8s.io/apiextensions-apiserver/pkg/apiserver/helpers.go

```diff
@@ -37,6 +38,57 @@ func getColumnsForVersion(crd *apiextensionsv1.CustomResourceDefinition, version
 	return nil, fmt.Errorf("version %s not found in apiextensionsv1.CustomResourceDefinition: %v", version, crd.Name)
 }
 
+// getStatusColumnsForVersion returns the columns under the status section for given version or nil.
+// NOTE: the newly logically-defaulted columns is not pointing to the original CRD object.
+// One cannot mutate the original CRD columns using the logically-defaulted columns. Please iterate through
+// the original CRD object instead.
+func getStatusColumnsForVersion(crd *apiextensionsv1.CustomResourceDefinition, version string) ([]apiextensionsv1.CustomResourceColumnDefinition, error) {
+	for _, v := range crd.Spec.Versions {
+		if version == v.Name {
+			// filter the columns to only those with json paths starting with ".status"
+			var cols []apiextensionsv1.CustomResourceColumnDefinition
+			for _, col := range v.AdditionalPrinterColumns {
+				if strings.HasPrefix(col.JSONPath, ".status") {
+					cols = append(cols, col)
+				}
+			}
+			return serveDefaultColumnsIfEmpty(cols), nil
+		}
+	}
+	return nil, fmt.Errorf("version %s not found in apiextensionsv1.CustomResourceDefinition: %v", version, crd.Name)
+}
+
+// getScaleColumnsForVersion returns 2 columns.
+// One for desired state and the other for actual statue
+func getScaleColumnsForVersion(crd *apiextensionsv1.CustomResourceDefinition, version string) ([]apiextensionsv1.CustomResourceColumnDefinition, error) {
+	for _, v := range crd.Spec.Versions {
+		if version == v.Name {
+			// filter the columns to only those with json paths starting with ".status"
```

Comment:
This comment is not valid here, right? Because we are adding a column based on `.spec.replicas`?